flink-table/flink-table-planner-blink/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -2670,12 +2670,12 @@ private boolean accept_(RexNode e, List<RexNode> newTerms) {
                             e.getKind(),
                             newTerms);
                 case IS_NULL:
-                    if (negate) {
-                        return false;
-                    }
-                    final RexNode arg = ((RexCall) e).operands.get(0);
-                    return accept1(
-                            arg, e.getKind(), rexBuilder.makeNullLiteral(arg.getType()), newTerms);
+                    /**
```

Comment:
We should also add comment in the class javadoc to mention what has been changed, and when this file can be removed until which calcite issue is fixed (we should create a corresponding issue in Calcite first).